src/rascal/structure_managers/species_manager.hh

```diff
@@ -249,21 +249,23 @@ namespace rascal {
     /**
      * Helper struct that loops over a cluster or manager, and
      * segregates the iteratee (i.e. the next higher order clusters)
-     * by species. If the loop has not reached the highest cluster
-     * order (i.e. MaxOrder), it recursively loops also over the next
-     * higher order clusters.
+     * by species.
      */
     template <class ManagerImplementation, size_t MaxOrder,
-              size_t Remaining = MaxOrder>
+              size_t CurrentOrder = 2, bool Stop = false>
     struct FilterSpeciesLoop {
       using SpeciesManager_t = SpeciesManager<ManagerImplementation, MaxOrder>;
+      static constexpr bool StopNextOrder{MaxOrder == CurrentOrder};
+
       using NextFilterSpeciesLoop =
-          FilterSpeciesLoop<ManagerImplementation, MaxOrder, Remaining - 1>;
+          FilterSpeciesLoop<ManagerImplementation, MaxOrder, CurrentOrder + 1,
```

Comment:
it used to start at order ==1 up to order == MaxOrder. now it does order ==1 separately and starts at order == 2 up to order == MaxOrder. The logic has not changed fundamentally I would say just Remaining was not useful anymore so I switch it with CurrentOrder so that I can use get_clusters_of_order<CurrentOrder>() .

Reply:
So I guess this is related to the comment above in the sense that we do not need the intermediate orders any more, because we are able to jump directly?
I have to think about this. I am wondering if we lose the intermediate orders  - which in the new logic might not even exist.